Multisig/UI/WalletConnectV2/WalletConnectManager.swift

```diff
@@ -73,140 +73,184 @@ class WalletConnectManager {
         guard let uri = WalletConnectURI(string: url) else { return }
         dappConnectedTrackingEvent = trackingEvent
         pairClient(uri: uri)
     }
-
+    
     func pairClient(uri: WalletConnectURI) {
         Task {
             do {
-                try await Sign.instance.pair(uri: uri)
+                // Clean up pairings to prevent pairingAlreadyExist error
+                disconnectUnusedPairings()
+                try await Web3Wallet.instance.pair(uri: uri)
                 NotificationCenter.default.post(name: .wcConnectingSafeServer, object: self)
             } catch {
                 LogService.shared.error("DAPP: Failed to register to remote notifications \(error)")
+                Task { @MainActor in
+                    App.shared.snackbar.show(message: "DAPP: Failed to register to remote notifications: \(error)")
+                }
             }
         }
     }
-
+    
     private func sign(request: Request, response: AnyCodable) {
         Task {
             do {
-                try await Sign.instance.respond(topic: request.topic, requestId: request.id, response: .response(response))
+                try await Web3Wallet.instance.respond(topic: request.topic, requestId: request.id, response: .response(response))
             } catch {
                 print("DAPP: Respond Error: \(error.localizedDescription)")
+                Task { @MainActor in
+                    App.shared.snackbar.show(message: "DAPP: Respond Error: \(error)")
+                }
             }
         }
     }
-
+    
     func reject(request: Request) {
         Task {
             do {
-                try await Sign.instance.respond(
+                try await Web3Wallet.instance.respond(
                     topic: request.topic,
                     requestId: request.id,
                     response: .error(.init(code: 0, message: ""))
                 )
             } catch {
                 print("DAPP: Respond Error: \(error.localizedDescription)")
+                Task { @MainActor in
+                    App.shared.snackbar.show(message: "DAPP: Respond Error: \(error.localizedDescription)")
+                }
             }
         }
     }
-
+    
     func approveSession(proposal: Session.Proposal) {
         Task {
             guard let safe = try? Safe.getSelected() else { return }
+            
             var sessionNamespaces = [String: SessionNamespace]()
             proposal.requiredNamespaces.forEach {
                 let caip2Namespace = $0.key
                 let proposalNamespace = $0.value
-                let chains = proposalNamespace.chains
-
-                let accounts = Set(chains.compactMap {
+                guard let chains = proposalNamespace.chains else { return }
+                
+                let selectedSafeChain = chains.filter { chain in
+                    chain.namespace == "eip155" && chain.reference == safe.chain?.id
```

Comment:
Can we please just extract this magic value to constant variable "eip155"?

Reply:
Do we have a good place for this? Or can I just define it in this file?